dashboard/app/models/course_version.rb

```diff
@@ -151,6 +152,25 @@ def recommended?(locale_code = 'en-us')
     latest_stable_version == content_root
   end
 
+  def self.courses_for_unit_selector(unit_ids, user)
```

Comment:
I think some comments here describing the algorithm for assembling the list would be helpful.  Include some background on _why_ single unit course offerings are handled differently than unit group course versions.

Btw, why do we ask CourseOffering for single unit courses but CourseVersion for unit group courses?

Reply:
Good call. I will add comments.

We use Course Offerings for single unit course offerings because we want to group together all the course offerings across years. So all the Course A's are together under the Course A header. Where as for unit group courses we want to have all the units for a specific year grouped together under the unit group for that year. So CSD has multiple headers in the list with the units for that year under it.